server/api/reviews.go

```diff
@@ -17,8 +22,63 @@ type ReviewBody struct {
 	Categories []string `json:"categories"`
 }
 
-func getAll() ([]*db.Review, error) {
-	return utils.FilterReviews(bson.D{{}})
+func getAll(sortQuery string, searchQuery string) ([]db.Review, error) {
+	var reviewsCursor *mongo.Cursor
+	var err error
+
+	DB, err := db.DB()
+
+	if err != nil {
+		return nil, err
+	}
+
+	// close db connection
+	defer func() {
+		err := DB.Client().Disconnect(db.Ctx)
+		if err != nil {
+			return
+		}
+	}()
+
+	opts := options.Find()
+
+	if len(sortQuery) > 0 {
+		recencySort, err := utils.ConvertStringToNum(sortQuery)
+
+		if err != nil {
+			return nil, errors.New("Invalid value for sort query")
+		}
+
+		if recencySort == 1 {
+			opts.SetSort(bson.D{{Key: "likes", Value: 1}})
+		} else {
+			opts.SetSort(bson.D{{Key: "likes", Value: -1}})
+		}
+
+	} else {
+		opts.SetSort(bson.D{{Key: "likes", Value: -1}})
+	}
+
+	matchKeyWord := bson.D{{Key: "book_name", Value: bson.D{{
+		Key: "$regex", Value: primitive.Regex{Pattern: fmt.Sprintf("^%s.*", searchQuery), Options: "i"}}}}}
+
+	if len(searchQuery) > 0 {
+		reviewsCursor, err = DB.Collection(db.ReviewsCollection).Find(db.Ctx, matchKeyWord, opts)
```

Comment:
The search doesn't meet the requirement in this card: https://github.com/codeday-labs/bookworms/projects/1#card-64509965

As the result is not sorted by the number of matched keywords in this case.

For example:
review1 = 'An apple a day keeps the doctor away', 3 likes, created at July 19
review2 = 'I like to have an apple, because I like apple, especially green apples', 2 likes, created at July 1
review3 = 'Instead of apple, I like orange, banana and kiwi more', 1 likes, created at July 20

If the user searches 'apple', the order will be
- review2 (# of apple = 3)
- review1 (# of apple = 1, 3 likes)
- review3 (# of apple = 1, 1 like)
> If user is sorting by recency, the order will be review2 -> review3 -> review1 instead

If the user searches 'apple orange banana kiwi', the order will be
- review3 (1 apple + 1 orange + 1 banana + 1 kiwi = 4)
- review2 (3 apple)
- review1 (1 apple)